ext/nnstreamer/android_source/gstamcsrc.c

```diff
@@ -0,0 +1,971 @@
+      if (res < 0) {
+        priv->seekable = FALSE;
+        LOGE ("Error: lseek() failed");
+        goto error_close;
+      }
+      priv->seekable = TRUE;
+    }
+  }
+
+  priv->seekable = priv->seekable && priv->is_regular;
+
+  return gst_amc_src_codec_config (self);
+
+error_close:
+  close (priv->fd);
+error:
+  return FALSE;
+}
+
+/**
+ * @brief set file location for media codec
+ * @param[in] location Name of the media file
+ */
+static gboolean
+gst_amc_src_set_location (GstAMCSrc * self, const gchar * location,
+    GError ** err)
+{
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  if (location == NULL) {
+    priv->filename = NULL;
+  } else {
+    priv->filename = g_strdup (location);
+    return gst_amc_src_media_open (self);
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief set amcsrc properties
+ */
+static void
+gst_amc_src_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  GstAMCSrc *self = GST_AMC_SRC (object);
+
+  switch (prop_id) {
+    case PROP_LOCATION:
+      gst_amc_src_set_location (self, g_value_get_string (value), NULL);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief get amcsrc properties
+ */
+static void
+gst_amc_src_get_property (GObject * object, guint prop_id, 
+    GValue *value, GParamSpec * pspec)
+{
+  GstAMCSrc *self = GST_AMC_SRC (object);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  switch (prop_id) {
+    case PROP_LOCATION:
+      g_value_set_string (value, priv->filename);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief unlock function, flush any pending data in the data queue
+ */
+static gboolean
+gst_amc_src_unlock (GstBaseSrc * src)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  LOGI ("unlocking create");
+  gst_data_queue_set_flushing (priv->outbound_queue, TRUE);
+
+  return TRUE;
+}
+
+/**
+ * @brief unlock_stop function, clear the previous unlock request
+ */
+static gboolean
+gst_amc_src_unlock_stop (GstBaseSrc * src)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  LOGI ("stopping unlock");
+  gst_data_queue_set_flushing (priv->outbound_queue, FALSE);
+
+  return TRUE;
+}
+
+/**
+ * @brief get caps of subclass
+ */
+static GstCaps *
+gst_amc_src_get_caps (GstBaseSrc * src, GstCaps * filter)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+  GstCaps *caps = gst_caps_new_empty ();
+  GstStructure *format;
+
+  GST_OBJECT_LOCK (self);
+
+  /** width, height, and framerate were obtained from the media format */
+  format = gst_structure_new ("video/x-raw",
+      "format", G_TYPE_STRING, "NV12",  /** TODO Support other formats? */
```

Comment:
You can set the caps in the template, and simply use `gst_pad_get_pad_template_caps (pad)` or `gst_pad_get_current_caps (pad)`

Reply:
Hmm. I just wanted to notify available caps (obtained from the media format of an input file) to other elements.
Is it possible to perform the same behavior using your suggestion?

Reply:
Yes, you are right. Cannot set in the template.
You can consider taking intersection of the new caps formed here with gst_pad_get_current_caps (pad), if it is not NULL.